ibis/expr/operations/temporal.py

```diff
@@ -99,7 +97,7 @@ class DateTruncate(Value):
 @public
 class TimeTruncate(Value):
     arg = rlz.time
-    unit = rlz.map_to(_time_units)
+    unit = rlz.coerced_to(TimeUnit)
```

Comment:
Where we use `unit = rlz.coerced_to(TimeUnit)` we can switch to type annotations `unit: TimeUnit`